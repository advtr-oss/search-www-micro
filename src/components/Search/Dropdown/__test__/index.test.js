/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import { Dropdown } from '../index'

const noop = () => {}

describe('Search.Dropdown', function () {
  it('should display the container', function () {
    const { container } = render(<Dropdown values={[]} onSelect={noop} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display the loading container', function () {
    const { container } = render(<Dropdown values={[]} onSelect={noop} loading />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
