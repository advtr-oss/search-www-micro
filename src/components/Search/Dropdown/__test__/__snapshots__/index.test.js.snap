// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.Dropdown should display the container 1`] = `
.c0 > ul {
  padding: 0;
  margin: 0;
  position: absolute;
  width: 100%;
  z-index: 1000;
  top: 100%;
  display: block;
  overflow: hidden;
}

.c0 > ul:before {
  content: attr(data-title);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  border-bottom: 1px solid;
  font-size: 12px;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
}

.c0 > ul > li:not(:last-child) {
  border-color: var(--border);
  border-bottom: 1px solid;
}

.c1 {
  width: 100%;
}

<div
  class="c0"
>
  <ul
    class="c1"
    data-title="Search Results"
    role="listbox"
  />
</div>
`;

exports[`Search.Dropdown should display the loading container 1`] = `
.c0 > ul {
  padding: 0;
  margin: 0;
  position: absolute;
  width: 100%;
  z-index: 1000;
  top: 100%;
  display: block;
  overflow: hidden;
}

.c0 > ul:before {
  content: attr(data-title);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  border-bottom: 1px solid;
  font-size: 12px;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
}

.c0 > ul > li:not(:last-child) {
  border-color: var(--border);
  border-bottom: 1px solid;
}

.c1 {
  width: 100%;
}

.c2 {
  height: 48px;
  padding: 0 15px;
  display: grid;
  grid-auto-flow: column;
  grid-template-columns: max-content;
  grid-gap: 15px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  line-height: 16px;
}

.c3 {
  -webkit-animation: bQashm 1.5s ease-in-out infinite;
  animation: bQashm 1.5s ease-in-out infinite;
  color: transparent;
  cursor: progress;
  display: block;
  width: 100%;
  height: 100%;
}

.c4 {
  width: 24px;
  height: 24px;
}

.c5 {
  height: 24px;
  width: 100%;
}

<div
  class="c0"
>
  <ul
    class="c1"
    data-title="Search Results"
    role="listbox"
  >
    <li
      role="option"
      tabindex="-1"
    >
      <div
        class="c2 loading-item"
      >
        <span
          class="c3 c4"
        />
        <span>
          <span
            class="c3 c5"
          >
            loading...
          </span>
        </span>
      </div>
    </li>
    <li
      role="option"
      tabindex="-1"
    >
      <div
        class="c2 loading-item"
      >
        <span
          class="c3 c4"
        />
        <span>
          <span
            class="c3 c5"
          >
            loading...
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
